src/core/Form/Textarea/Textarea.tsx

```diff
@@ -1,52 +1,162 @@
   error: `${baseClassName}--error`,
 };
 
-export interface TextareaProps extends CompTextareaProps, TokensProp {}
+type TextareaStatus = Exclude<InputStatus, 'success'>;
 
-const StyledTextarea = styled(
-  ({ tokens, ...passProps }: TextareaProps & InternalTokensProp) => (
-    <CompTextarea {...passProps} />
-  ),
-)`
-  ${(props) => baseStyles(props)}
-`;
+export interface TextareaProps extends HtmlTextareaProps, TokensProp {
+  /** Custom classname to extend or customize */
+  className?: string;
+  /** Disable usage */
+  disabled?: boolean;
+  /** Event handler to execute when clicked */
+  onClick?: () => void;
+  /** To execute on textarea text change */
+  onChange?: (event: ChangeEvent<HTMLTextAreaElement>) => void;
+  /** To execute on textarea text onBlur */
+  onBlur?: (event: FocusEvent<HTMLTextAreaElement>) => void;
+  /** Label */
+  labelText: string;
+  /** Hide or show label. Label element is always present, but can be visually hidden.
+   * @default visible
+   */
+  labelMode?: 'hidden' | 'visible';
+  /** Placeholder text for input. Use only as visual aid, not for instructions. */
+  visualPlaceholder?: string;
+  /** Text content for textarea */
+  children?: string;
+  /** Hint text to be shown below the component */
+  hintText?: string;
+  /**
+   * 'default' | 'error'
+   * @default default
+   */
+  status?: TextareaStatus;
+  /** Status text to be shown below the component and hint text. Use e.g. for validation error */
+  statusText?: string;
+  /** Resize mode of the textarea
+      'both' | 'vertical' | 'horizontal' | 'none'
+      @default 'vertical' 
+   */
+  resize?: 'both' | 'vertical' | 'horizontal' | 'none';
+  /** Optional text that is shown after labelText. Will be wrapped in parentheses. */
+  optionalText?: string;
+  /**
+   * Unique id
+   * If no id is specified, one will be generated automatically
+   */
+  id?: string;
+  /** Input name */
+  name?: string;
+}
 
-export class Textarea extends Component<TextareaProps> {
+class BaseTextarea extends Component<TextareaProps> {
   render() {
     const {
-      children,
+      id,
+      className,
       disabled = false,
+      children,
+      onClick,
+      labelMode,
+      labelText,
+      hintText,
       status,
-      className,
+      statusText,
+      visualPlaceholder,
+      resize: dismissResize,
+      optionalText,
+      'aria-describedby': ariaDescribedBy,
       ...passProps
-    } = withSuomifiDefaultProps(this.props);
+    } = this.props;
+
+    const onClickProps = !!disabled ? {} : { onMouseDown: onClick };
+    const statusTextId = `${id}-statusText`;
+    const hintTextId = `${id}-hintText`;
+
+    const getDescribedBy = () => {
+      if (statusText || hintText || ariaDescribedBy) {
+        return {
+          'aria-describedby': [
+            ...(statusText ? [statusTextId] : []),
+            ...(hintText ? [hintTextId] : []),
+            ...(ariaDescribedBy ? [ariaDescribedBy] : []),
```

Comment:
This is purely cosmetic, but in case there is only one element in the array, resulting string will contain an empty space after the id. I'd propose we write an ariaIdsTool to do this work nicely. So that you can pass in an array of string | undefined and the tool returns you a string.

Reply:
The tool should be used in multiple places already, not worth doing for only this use case.

Reply:
This is now done with the util located `src/utils/aria.ts`